openc3/lib/openc3/models/microservice_model.rb

```diff
@@ -237,7 +239,9 @@ def undeploy
       @bucket.list_objects(bucket: ENV['OPENC3_CONFIG_BUCKET'], prefix: prefix).each do |object|
         @bucket.delete_object(bucket: ENV['OPENC3_CONFIG_BUCKET'], key: object.key)
       end
-      ConfigTopic.write({ kind: 'deleted', type: 'microservice', name: @name, plugin: @plugin }, scope: @scope)
+      config = { kind: 'deleted', type: 'microservice', name: @name }
+      config[:plugin] = @plugin if @plugin
+      ConfigTopic.write(config, scope: @scope)
```

Comment:
Turns out this is a bug if you don't have a plugin defined. Throws `Unsupported command argument type: NilClass` when you try to write the config topic. The exception is way down in the xadd part of the redis streams.